common/src/main/java/org/opensearch/ml/common/model/MLModelConfig.java

```diff
@@ -33,13 +38,33 @@ public MLModelConfig(String modelType, String allConfig) {
         this.allConfig = allConfig;
     }
 
+    public MLModelConfig(String modelType, String allConfig, Map<String, Object> additionalConfig) {
```

Comment:
Hmm, you already have a new BaseModelConfig extending this abstract class to add a new additionalConfig. I am not sure what the difference is for these two classes. I think you don't have to add additionalConfig here that causes a version control for the existing interface change. We keep this interface unchanged, and have like TextEnbeddingModelConfig extending your new BaseModelConfig that would have less impact.